lms/djangoapps/teams/serializers.py

```diff
@@ -160,3 +162,6 @@ def __init__(self, *args, **kwargs):
         topics_to_team_count = {d['topic_id']: d['team_count'] for d in teams_per_topic}
         for topic in self.data['results']:
             topic['team_count'] = topics_to_team_count.get(topic['id'], 0)
+
+        if 'sort_order' in self.context and self.context['sort_order'] == 'team_count':
+            list.sort(self.data['results'], key=lambda t: t['team_count'], reverse=True)
```

Comment:
Minor nit: could you use `self.data['results'].sort( ... )` here? I usually find that notation a bit more readable.